SWLOR.Game.Server/Native/ResolveAttackRoll.cs

```diff
@@ -24,7 +28,366 @@ public static void RegisterHook()
+                if (weapon != null)
+                {
+                    attackOverride = weapon.m_ScriptVars.GetInt(new CExoString("ATTACK_TYPE_OVERRIDE"));
+
+                    if (attackOverride != 0 & attackOverride < 4)
+                    {
+                        attackType = (uint)attackOverride;
+                    }
+                    else if (pAttackData.m_bRangedAttack == 1 && attacker.GetRangeWeaponEquipped() == 1)
+                    {
+                        attackType = (uint) AttackType.Ranged;
+                    }
+                }
+            }
+
+            Log.Write(LogGroup.Attack, "Selected attack type " + attackType.ToString() + ", weapon " + (weapon == null ? "none":weapon.GetFirstName().GetSimple(0)) );
+
+            // We now have our attack type defined.  Pull the relevant attributes, defaulting to melee.
+            int attackAttribute = attackerStats.m_nStrengthModifier;
+            int defendAttribute = defenderStats.m_nStrengthModifier;
+
+            switch (attackType)
+            {
+                case (uint)AttackType.Ranged:
+                    attackAttribute = attackerStats.m_nDexterityModifier;
+                    defendAttribute = defenderStats.m_nDexterityModifier;
+                    break;
+                case (uint)AttackType.Spirit:
+                    attackAttribute = attackerStats.m_nWisdomModifier;
+                    defendAttribute = defenderStats.m_nWisdomModifier;
+                    break;
+            }
+
+            // Check for negative modifiers.  A modifier of -2 is represented as 254.
+            if (attackAttribute > 128) attackAttribute -= 256;
+            if (defendAttribute > 128) defendAttribute -= 256;
+
+            Log.Write(LogGroup.Attack, "Attacker attribute modifier: " + attackAttribute.ToString() +", defender attribute modifier: " + defendAttribute.ToString());
+
+            //---------------------------------------------------------------------------------------------
+            //---------------------------------------------------------------------------------------------
+            //---------------------------------------------------------------------------------------------
+            // Modifiers - put in modifiers here based on the type of attack (and type of weapon etc.).
+            int modifiers = 0;
+
+            // Defender stunned
+            if (defender.m_nState == 6) // Stunned
+            {
+                defendAttribute = 0;
+            }
+
+            // Defender not targeting the attacker.
+            // Dev note: the GetItem method always creates a new instance of CNWActionNode so there should be no NPEs.
+            // Note: this always returns object invalid for NPCs (2130706432) as their actions aren't represented the same way.
+            uint oidTarget = defender.m_pActionQueue.GetItem(0).oidTarget;
+
+            if (oidTarget == 2130706432) // OBJECT_INVALID
```

Comment:
In this case I think it's safe for you to use NWScript.OBJECT_INVALID instead of hard coding the Id here. It's about the only thing you can safely reference since it's a constant uint value. If it's not possible, can you assign it as a constant in this file by itself?

Reply:
Good call - should have figured this was already defined somewhere.   Updated (and retested).